src/sdk/namespace_fs.js

```diff
@@ -2912,28 +2935,35 @@ class NamespaceFS {
         return path.normalize(path.join(versions_dir_path, key_version));
     }
 
-    // this function returns the following version information -
-    // version_id_str - mtime-{mtimeNsBigint}-ino-{ino} | explicit null
-    // mtimeNsBigint - modified timestmap in bigint - last time the content of the file was modified
-    // ino - refers to the data stored in a particular location
-    // delete_marker - specifies if the version is a delete marker
-    // path - specifies the path to version
-    // if version xattr contains version info - return info by xattr
-    // else - it's a null version - return stat
-    // if fd is passed, will use fd instead of path to stat
-    async _get_version_info(fs_context, version_path, fd) {
+
+    /**
+     * this function returns the following version information -
+     * version_id_str - mtime-{mtimeNsBigint}-ino-{ino} | explicit null
+     * mtimeNsBigint - modified timestmap in bigint - last time the content of the file was modified
+     * ino - refers to the data stored in a particular location
+     * delete_marker - specifies if the version is a delete marker
+     * path - specifies the path to version
+     * if version xattr contains version info - return info by xattr
+     * else - it's a null version - return stat
+     * if file is passed, will use file instead of path to stat
+     * @param {nb.NativeFSContext} fs_context 
+     * @param {String} version_path 
+     * @param {nb.NativeFile} [file] 
+     * @returns {Promise<object>}
+     */
+    async _get_version_info(fs_context, version_path, file = undefined) {
         try {
-            const stat = fd ? await fd.stat(fs_context, { skip_user_xattr: true }) :
-                await nb_native().fs.stat(fs_context, version_path, { skip_user_xattr: true });
-            dbg.log1('NamespaceFS._get_version_info stat ', stat, version_path, fd);
+            const stat = file ? await file.stat(fs_context) : await nb_native().fs.stat(fs_context, version_path);
+            dbg.log1('NamespaceFS._get_version_info stat ', stat, version_path, file);
 
             const version_id_str = this._get_version_id_by_xattr(stat);
             const ver_info_by_xattr = this._extract_version_info_from_xattr(version_id_str);
             return {
                 ...(ver_info_by_xattr || stat),
                 version_id_str,
                 delete_marker: stat.xattr[XATTR_DELETE_MARKER],
-                path: version_path
+                path: version_path,
+                stat
```

Comment:
its just my opinion. but adding the stat here doesn't seem like the proper way of passing the stat. this function is used to get version specific information, some of it from the stat itself. this function is used in many places. adding the stat here for a specific purpose seems kind of forced, especially if we have both the stat and the stats content in the same object

Reply:
I did this because I saw multiple stat calls for MISMATCH VERSION after calls to this function 
these stat calls are redundant file system calls so I wanted to refactor it, 
I agree that we should do it on a different PR though
also the same for checking MISMATCH on posix scenario will use this

Reply:
so I removed it for now and called stat again when needed as we do in MISMATCH VERSION check
we should still fix it on a different PR though